go/ngql/types.go

```diff
@@ -196,13 +196,15 @@ func structToGQLObject(nomsType *types.Type, tm *typeMap) *graphql.Object {
 			}
 
 			structDesc.IterFields(func(name string, nomsFieldType *types.Type) {
-				fieldType := NomsTypeToGraphQLType(nomsFieldType, false, tm)
+				fieldType := unpackNonNullType(NomsTypeToGraphQLType(nomsFieldType, false, tm))
```

Comment:
This `unpackNonNullType` thing doesn't feel right. It makes me wonder if we should do a sweep and *not* mark every type as non-null in the first place, then push that decision into the containers or wherever the fields are stored.

I.e. I'm looking at `NomsTypeToGraphQLType` which wraps everything in `graphql.NewNonNull`. What if:
* We didn't do that.
* We add the non-nullability into the list/set/map keys/values and ref.

Perhaps the problem is that graphql conflates optionality with nullability, but oh well.